### ISSUE TITLE
Prompt user to select an ESLint based lint/format config (eg Airbnb) like Vue CLI

### ISSUE THREAD

**Issue by lloydjatkinson** [vuejs/create-vue#82]
Hi,

Thank you for creating this tool!

It would be really great if `create-vue` could setup and configure (including with JS *or* TS support) a chosen ruleset - like Vue CLI does.

For example when using Vue CLI a choice is presented:

![image](https://user-images.githubusercontent.com/5633938/160260799-c5a1abe6-2a4b-4df9-ab5d-0ab914c0f78e.png)

`vue-create` today:

![image](https://user-images.githubusercontent.com/5633938/160260859-c0dd0103-7795-4acd-af12-88a9897d1e02.png)

I have looked into how this can be setup and honestly it's pretty hard to understand - ESLint is powerful but hard to configure **correctly**. There appears to be many possible solutions but once Typescript is in the mix the complexity increases for anyone, like me, that isn't that familiar with the really low level details of configuring ESLint. Even if someone is able to work through the (many) conflicting articles on setting this up there is still the reality of needing to go through the same complexity for every new project.

One of the great features of Vue CLI is that this is already configured for a user if they pick TS, ESLint, Aibnb.

Please could this prompt be added to `create-vue`? (and the `Standard` standard too for users of that)

**Comment by Reidond** [comment]
it's not that hard to bootstrap eslint config on your own, you could even make your own packages with predefined config.
i personally use eslint + prettier combo and it works great

**Comment by Reidond** [comment]
### Packages
```json
{
    "eslint": "^8.10.0",
    "eslint-config-prettier": "^8.4.0",
    "eslint-plugin-prettier": "^4.0.0",
    "eslint-plugin-vue": "^8.5.0",
    "prettier": "2.5.1",
    "vue-eslint-parser": "^8.3.0"
}
```
### Configs
.eslintrc.js
```js
module.exports = {
  root: true,
  env: {
    browser: true,
    node: true,
    es6: true,
    "vue/setup-compiler-macros": true,
  },
  parser: "vue-eslint-parser",
  parserOptions: {
    sourceType: "module",
    ecmaVersion: "latest",
  },
  extends: [
    "eslint:recommended",
    "plugin:vue/vue3-recommended",
    "plugin:prettier/recommended",
  ],
  rules: {
    "vue/multi-word-component-names": "off",
  },
};
```
.prettierrc.json
```json
{
  "trailingComma": "all",
  "tabWidth": 2,
  "semi": true,
  "singleQuote": false
}
```
.editorconfig
```editorconfig
root = true

[*]
indent_style = space
indent_size = 2
charset = utf-8
trim_trailing_whitespace = true
insert_final_newline = true
end_of_line = lf
```

**Comment by lloydjatkinson** [comment]
And you don’t see the value in having this already exist as an option as per what I said about Vue CLI?

**Comment by Reidond** [comment]
Well from my experience you can't rely on those options because they can be outdated when packages update (and they will update often) and you need to know how to setup this manually, it's really not hard.

You don't have to use examples from internet, use eslint documentation

**Comment by Reidond** [comment]
I think `create-vue` should be very simple to setup without unnecessary junk that people use, not knowing how to update it when things go wrong

**Comment by LilPidgey** [comment]
> you need to know how to setup this manually, it's really not hard.

This is such a stuck up take. Of course, you may have experience as a JS developer, but what about the people dipping their toes into Vue as their first framework? They'll be repulsed by the fact that the basic project setup does not even properly work. It's about ease of use and not  knowing or not knowing. As mentioned, Vue CLI is able to do this just fine. There is no reason this shouldn't be able to.

**Comment by lloydjatkinson** [comment]
This is an extremely poor take on the situation. Just because *you* don’t use it doesn’t make xyz “junk”. Vue CLI supports setting this up.

“read the ESLint documentation” yes don’t you think I did that before creating the issue? The point you seem to have missed is that not everyone can be an expert at every tool in a stack and it would be good to have certain tooling that’s expected in a modern stack to be an option.

With your argument, create-vue may as well not exist because “it’s not that hard” to set everything up.

I’m not really sure what the point of this was, if you aren’t willing to contribute to the use case in a meaningful way.

**Comment by AisonSu** [comment]
Could i will a contribution with this feature?

**Comment by AisonSu** [comment]
Could i give a contribution with this feature request?
我可以提交一个新PR以提供这个特性吗？
@sodatea

**Comment by lloydjatkinson** [comment]
Is there no plan on implementing this?

**Comment by sodatea** [comment]
My plan:
- [x] new `standard` configs
- [ ] new `airbnb` configs, almost finished, plan to release on the next workday
- [ ] check if the `typescript` config needs further refactoring
- [ ] integrate these configs into a `@vue/create-eslint-config` package
- [ ] integrate that package into `create-vue`

**Comment by AisonSu** [comment]
so the renderEslint will be transformed into a new package?

**Comment by sodatea** [comment]
Yeah, so that we can have better tests for its functionality and support Vue CLI projects.

**Comment by sodatea** [comment]
I'll post a link here after the initial prototype. Contribution to tests would be much appreciated.

**Comment by AisonSu** [comment]
looking forward to it very much

**Comment by JFGHT** [comment]
Any update on this?

**Comment by haoqunjiang** [comment]
While I personally still intend to work on `@vue/eslint-config-airbnb` (working on it lately, though not finished), I find it harder and harder to keep it up-to-date with the broader ecosystem. It has too wide a scope, yet the maintainers of the upstream package(s) do not prioritize new feature adoptions (as can be seen in their issue discussions).

I would not want to provide an often out-of-date config for `create-vue` users, so I'm closing this issue as "won't fix".

As for the [`standard` style guide](https://github.com/vuejs/eslint-config-standard), it's much easier to maintain and is already up-to-date. But it's facing similar problems: the upstream development has almost stalled. For example, its flat config PR was [merged
nearly two years ago](https://github.com/standard/eslint-config-standard/pull/316) but never got released. I had to copy-paste its source code into `@vue/eslint-config-standard`, which isn't really a good idea in the long term. So I don't feel like recommending it to average users, either.
